src/components/content/injection.tsx

```diff
@@ -28,27 +28,47 @@ export function Injection({ href }: InjectionProps) {
   useEffect(() => {
     const origin = window.location.host
     const protocol = window.location.protocol
-    void fetch(
-      `${protocol}//${origin}/api/frontend${encodeURI(
-        href.startsWith('/') ? href : `/${href}`
-      )}`
-    )
+    const encodedHref = encodeURI(href.startsWith('/') ? href : `/${href}`)
+
+    try {
+      const cachedData = sessionStorage.getItem(encodedHref)
+      if (cachedData) {
+        dataToState(JSON.parse(cachedData))
+      }
+    } catch (e) {
+      //
+    }
+
+    void fetch(`${protocol}//${origin}/api/frontend${encodedHref}`)
       .then((res) => {
         if (res.headers.get('content-type')!.includes('json')) return res.json()
         else return res.text()
       })
-      .then((data: EntityProps) => {
-        if (data.contentType && data.data) {
-          setValue(data.data.value)
-          if (data.data.license) {
-            setLicense(data.data.license)
-          }
-          if (data.prettyLinks) {
-            setPrettyLinks(data.prettyLinks)
+      .then((fetchedData: EntityProps) => {
+        dataToState(fetchedData)
```

Comment:
... only set data when the injection is not loaded or data is different?

Reply:
Is it, though? If we always do the request anyways, there's not much point in caching, is there? We'd need at least a lifetime for the local cache, e.g. only fetch the data again if the data is older than 10 minutes or whatever.

Reply:
the point of caching is immediate response without waiting/back-navigation hickups. it would even make sense if the do the request anyways. 
but to be honest I just forgot the `return` 😄

Reply:
Okay, true. Still: we'd should optimize the requests, too, so that our site is also fast for mobile users (where additional requests would make other requests - where we don't have the data in cache - slower and waste mobile data).

Reply:
@inyono totally agree, my point was, that the cache makes sense, not that unnecessary requests do.

Let's talk about this again when we build a service worker (#347) or when we think about the session-cache for logged in users (edits and whatnot).

Reply:
(missing return is added in ca3f796bc9a60628ca4dc1b58045c9555953ecc6